src/Grid/mixed_grid.jl

```diff
@@ -0,0 +1,51 @@
+mutable struct MixedGrid{dim, C, T<:Real} <: Ferrite.AbstractGrid{dim}
+    cells::C # Tuple of concretely typed cell vectors
+    ncells_per_vector::Vector{Int}
+    nodes::Vector{Node{dim,T}}
+    # Sets
+    cellsets::Dict{String,Set{Int}}
+    nodesets::Dict{String,Set{Int}}
+    facesets::Dict{String,Set{FaceIndex}} 
+    edgesets::Dict{String,Set{EdgeIndex}} 
+    vertexsets::Dict{String,Set{VertexIndex}} 
+    # Boundary matrix (faces per cell × cell)
+    boundary_matrix::SparseMatrixCSC{Bool,Int}
```

Comment:
@termi-official is there a better way to encode the boundary information into the grid? Would be a good opportunity to include it in this struct

Reply:
Yes and no. It really depends on your use case. You can get away without using the `boundary_matrix` and I honestly think that it should not be in any grid struct, but in a separate (boundary information) data structure.

Reply:
I mean we have the facesets data structure anyway to distinguish different portions of the boundary - and a set lookup should be O(1). If we want to check if we are on the interior, then the upcoming topology PR should handle this better.

Reply:
Maybe @fredrikekre does know more.

Reply:
Note that removing the field is definitely a breaking change, so I do not think we can remove it immediately. (See https://github.com/Ferrite-FEM/Ferrite.jl/blob/ea716bc7a73da0b7970f81187b9b8b26ea502598/src/iterators.jl#L90 which is exported.) This is also why I asked Frederik for a comment.

Reply:
I guess the reason this exists is that it is faster than the set-lookup:
```julia
julia> grid = generate_grid(Triangle, (20, 20));

julia> ci = CellIterator(grid); reinit!(ci, 1);

julia> @btime onboundary($ci, $1);
  4.300 ns (0 allocations: 0 bytes)

julia> faceindex = FaceIndex(1,1)
FaceIndex((1, 1))

julia> @btime in($faceindex, $(grid.facesets["bottom"]))
  16.032 ns (0 allocations: 0 bytes)
```
So from that perspective, I'd suggest leaving it as it is for now and eventually tackle if it should be in the grid or not in another PR?

Reply:
It might make more sense to have it be a lookup for whether the cell has any face on a boundary. Then you can check once per cell, instead of once epr face. (This also works better with multi-celltype grids).